src/Sitecore.AspNetCore.SDK.RenderingEngine/Extensions/SitecoreFieldExtensions.cs

```diff
@@ -53,6 +132,80 @@ private static string GetSitecoreMediaUri(string url, object? imageParams)
             }
         }
 
+        return ApplyJssMediaUrlPrefix(url);
+    }
+
+    /// <summary>
+    /// Gets modified URL string to Sitecore media item with parameter preservation.
+    /// This method preserves existing URL parameters and merges them with new ones.
+    /// </summary>
+    /// <param name="urlStr">The URL string.</param>
+    /// <param name="parameters">Parameters to merge.</param>
+    /// <returns>Modified URL string.</returns>
+    private static string GetSitecoreMediaUriWithPreservation(string urlStr, object? parameters)
+    {
+        if (string.IsNullOrEmpty(urlStr))
+        {
+            return urlStr;
+        }
+
+        // Parse existing query parameters and build merged parameters dictionary
+        Dictionary<string, object?> mergedParams = new Dictionary<string, object?>(StringComparer.OrdinalIgnoreCase);
+        string url = ParseUrlParams(urlStr, mergedParams);
+
+        // Add new parameters (these will override existing ones)
+        AddParametersToResult(mergedParams, parameters, skipNullValues: true);
+
+        // Add query parameters
+        foreach (KeyValuePair<string, object?> kvp in mergedParams)
+        {
+            if (kvp.Value != null)
+            {
+                url = QueryHelpers.AddQueryString(url, kvp.Key, kvp.Value.ToString() ?? string.Empty);
+            }
+        }
+
+        return ApplyJssMediaUrlPrefix(url);
+    }
+
+    /// <summary>
+    /// Parses URL query string parameters and adds them to the provided dictionary.
+    /// </summary>
+    /// <param name="url">The full URL with potential query parameters.</param>
+    /// <param name="parameters">The dictionary to add parsed parameters to.</param>
+    /// <returns>The URL without query parameters.</returns>
+    private static string ParseUrlParams(string url, Dictionary<string, object?> parameters)
+    {
+        if (string.IsNullOrEmpty(url))
+        {
+            return url;
+        }
+
+        int queryIndex = url.IndexOf('?');
+        if (queryIndex < 0)
+        {
+            return url;
+        }
+
+        string cleanUrl = url.Substring(0, queryIndex);
+        string queryString = url.Substring(queryIndex);
+
+        Dictionary<string, Microsoft.Extensions.Primitives.StringValues> parsedQuery = QueryHelpers.ParseQuery(queryString);
+        foreach (KeyValuePair<string, Microsoft.Extensions.Primitives.StringValues> kvp in parsedQuery)
```

Comment:
Can you introduce a using statement for `Microsoft.Extensions.Primitives` so we don't have to fully qualify `StringValues` in both of these lines?